### DISCUSSION TITLE
How Do You Use the CLI?

### DISCUSSION THREAD

**Discussion by zicklag** [sonos/tract#422] [Q&A] [answered]
I didn't know there was a CLI until just now when I saw the release artifact. Is it possible to use the CLI to run the examples somehow, without even having to compile any Rust code? If so that would be cool to have instructions for in the examples.

**Comment by kali** [comment, marked as answer]
The command line is mostly meant for auditing networks, debugging issues, etc. My ML colleague also use it for visualizing, "flopping" (evaluate number of ops) or profiling nets. It is also used by the CI for running some full network tests. It may be worth writing a bit of documentation for it in the form of cooking recipes.

Feel free to play with it, I will answer questions too, but be aware of the following caveats :

* I don't consider the CLI interface stable, I want to be able to break it any time I need. As such, I strongly discourage its use it for "production scripting".
* Once again, documentation is poor: the best source is the --help parameter.
* Using it for running networks is possible, but often not really practical as data needs to be conditioned in a format the CLI can load. NPZ might be the best way to provide input, and I don't think we have any practical way to output results :)
* Finally, using the CLI to run network is terribly inefficient. Loading networks from ONNX and/or TF and getting them ready for tract is expensive... so I would not recommend it.

**Reply by zicklag** [reply]
OK, cool. Just curious what it was about and useful for. That answers my questions. :smiley:
